### PR TITLE
CRAM lossless roundtrip

### PR DESCRIPTION
**Pull request by vadimzalunin**
see https://github.com/samtools/htsjdk/issues/483 and https://github.com/samtools/htsjdk/issues/455

### CODE REVIEW

**Comment by vadimzalunin** [comment]
@cmnbroad please review
I'm limiting the tests to NM and MD only as they are the known offenders. Many of the roundtrip issues have been addressed previously in CRAMComplianceTest.

File: src/java/htsjdk/samtools/CRAMIterator.java [hunk 1]
```diff
@@ -49,7 +49,7 @@
     private ArrayList<SAMRecord> records;
     private SAMRecord nextRecord = null;
     @SuppressWarnings({"CanBeFinal", "FieldCanBeLocal"})
-    private boolean restoreNMTag = true;
+    private boolean restoreNMTag = false;
     @SuppressWarnings({"CanBeFinal", "FieldCanBeLocal"})
     private boolean restoreMDTag = false;
```
Comment (cmnbroad) [inline src/java/htsjdk/samtools/CRAMIterator.java]: Both restoreNMTag and restoreMDTag, and their corresponding SuppressWarnings annotations are no longer used and can be removed.

File: src/java/htsjdk/samtools/CRAMIterator.java [hunk 2]
```diff
@@ -201,8 +201,8 @@ void nextContainer() throws IOException, IllegalArgumentException,
                 final SAMSequenceRecord sequence = cramHeader.getSamFileHeader()
                         .getSequence(cramRecord.sequenceId);
                 refs = referenceSource.getReferenceBases(sequence, true);
-                if (samRecord.getReadBases() != SAMRecord.NULL_SEQUENCE)
-                    SequenceUtil.calculateMdAndNmTags(samRecord, refs, restoreMDTag, restoreNMTag);
+//                if (samRecord.getReadBases() != SAMRecord.NULL_SEQUENCE)
+//                    SequenceUtil.calculateMdAndNmTags(samRecord, refs, restoreMDTag, restoreNMTag);
```
Comment (cmnbroad) [inline src/java/htsjdk/samtools/CRAMIterator.java]: The commented out code can also be removed.

**Comment by cmnbroad** [comment]
I think this change and the corresponding test are sufficient for the specific issue in https://github.com/samtools/htsjdk/issues/455, but we shouldn't close https://github.com/samtools/htsjdk/issues/483 until we have much more comprehensive tests that exercise all of the SAMRecord features and use persisted BAM files. There are other issues that need to be resolved, i.e., see https://github.com/samtools/htsjdk/issues/499. So lets get this change in and we'll need to work on more comprehensive roundtrip tests in a separate PR.

**Comment by vadimzalunin** [comment]
@cmnbroad fixed but Travis failed due to some connectivity issue

**Pull request merged**